x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/recovery/SearchableSnapshotRecoveryState.java

```diff
@@ -51,6 +51,12 @@ public synchronized RecoveryState setRemoteTranslogStage() {
         return super.setStage(Stage.FINALIZE);
     }
 
+    @Override
+    public synchronized RecoveryState reset() {
+        setStage(Stage.INIT);
+        return this;
```

Comment:
Would it not be nicer to also return a new instance here? Would that be problematic?

Reply:
For searchable snapshots we shouldn't run into this problem, we don't recover any file through regular recovery. So I figured that it should be fine to reuse the same instance.

Reply:
To the reader however, it looks like this is either a micro-tuning (which seems unnecessary) or there is something special going on requiring us to not create a new state. I am fine to leave as is, but perhaps we can add a comment that it is equivalent to creating a new `SearchableSnapshotRecoveryState`?